### DISCUSSION TITLE
npm warn deprecated message

### DISCUSSION THREAD

**Discussion by RobinWitt** [neuefische/questions#512] [Web] [answered]
Hey everyone, while installing a new package to my capstone project, I got two npm warnings:

![Bildschirm­foto 2023-02-07 um 12 06 55](https://user-images.githubusercontent.com/117563487/217228635-87844cd4-f811-475d-961d-d8d2f41b2239.png)

Does that mean the `spotify-web-api-node` is using old versions of "formidable" and "superagent"?
Maybe you can give me a suggestion on whether I should ignore the messages for now or install the recommended versions. 
Thanks in advance!

**Comment by pixelass** [comment, marked as answer]
Last commit on that repo: [on Jan 24, 2021](https://github.com/thelinmichael/spotify-web-api-node/commit/be15f1c742b35134ce5bd35521d8bf1ab1ba67cf)

Looks like this package is not maintained anymore. You can try to install the latest "formidable" package and look into resolutions/overrides: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides


There's also an open issue about the status of that project: https://github.com/thelinmichael/spotify-web-api-node/issues/469

**Reply by RobinWitt** [reply]
Thanks for the reply, but I don't understand where I have to write the "overrides" arguments from the npm docs. 
Do I just change the version in `package-lock.json` in the specific `spotify-web-api-node` dependencies?

```jsx

"node_modules/spotify-web-api-node": {
      "version": "5.0.2",
      "resolved": "https://registry.npmjs.org/spotify-web-api-node/-/spotify-web-api-node-5.0.2.tgz",
      "integrity": "sha512-r82dRWU9PMimHvHEzL0DwEJrzFk+SMCVfq249SLt3I7EFez7R+jeoKQd+M1//QcnjqlXPs2am4DFsGk8/GCsrA==",
      "dependencies": {
        "superagent": "^6.1.0" // <- change this?
      }
    },

```

**Reply by pixelass** [reply]
it is a property on the root level of your `package.json`. 

```json
{
  "name": "capstone-project", 
  "dependencies": {
    "...": "all other dependencies",
    "formidable": "... what ever is latest that works"
  },
  "overrides": {
    "formidable": "... what ever is latest that works"
  }
}
```

after the adjust ment please delete the `package-lock.json` file and run `npm install`

**Comment by pixelass** [comment]
Aside from my other suggestion you can ignore this "warning". It means that there is no support for that version or support might not satisfy your expectations. 

These warnings are more importantin future-proof production apps. For your case you cvan ignore it but I suggest looking for security issues related to the two dependencies. You can also check if Spotify's API allows you to write your own serverless function that does the sae (pretty likely)

**Reply by RobinWitt** [reply]
Thanks a lot! I'm gonna consider ignoring the warnings for now or switching to another Spotify package.
